doc/plugins.dox

```diff
@@ -358,81 +464,79 @@ Once this is done, your first plugin is ready. Congratulations!
 @section listing Complete listing (without comments)
 
 @code
-#include <geanyplugin.h>
-
-GeanyPlugin		*geany_plugin;
-GeanyData		*geany_data;
-
-PLUGIN_VERSION_CHECK(211)
 
-PLUGIN_SET_INFO("HelloWorld", "Just another tool to say hello world",
-				"1.0", "John Doe <john.doe@example.org>");
-
-
-static GtkWidget *main_menu_item = NULL;
+#include <geanyplugin.h>
 
-static void item_activate_cb(GtkMenuItem *menuitem, gpointer gdata)
+static void item_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
 {
 	dialogs_show_msgbox(GTK_MESSAGE_INFO, "Hello World");
 }
 
-void plugin_init(GeanyData *data)
+
+static gboolean hello_init(GeanyPlugin *plugin, gpointer pdata)
 {
 	main_menu_item = gtk_menu_item_new_with_mnemonic("Hello World");
 	gtk_widget_show(main_menu_item);
 	gtk_container_add(GTK_CONTAINER(geany->main_widgets->tools_menu),
-		main_menu_item);
+			main_menu_item);
 	g_signal_connect(main_menu_item, "activate",
-		G_CALLBACK(item_activate_cb), NULL);
+			G_CALLBACK(item_activate_cb), NULL);
+
+	geany_plugin_set_data(plugin, main_menu_item, NULL);
+	retrun TRUE;
```

Comment:
On 18 August 2015 at 14:57, Thomas Martitz <notifications@github.com> wrote:

> In doc/plugins.dox
> <https://github.com/geany/geany/pull/469#discussion_r37264172>:
>
> >  	g_signal_connect(main_menu_item, "activate",
> > -		G_CALLBACK(item_activate_cb), NULL);
> > +			G_CALLBACK(item_activate_cb), NULL);
> > +
> > +	geany_plugin_set_data(plugin, main_menu_item, NULL);
> > +	retrun TRUE;
>
> I was wondering if an example source file containing the code from the
> docs could be included, that would be a starting point for new plugin
> makers and would test it works
> Great plan, let's call it demoplugin ;-)
>


​Neat, it just needs to be kept in sync then.​



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geany/geany/pull/469/files#r37264172>.
>